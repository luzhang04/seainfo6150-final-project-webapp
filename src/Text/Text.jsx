import React from "react";
import PropTypes from "prop-types";
import styles from "./Text.module.css";


const Text = (props) => {
  return <div className={styles.content}dangerouslySetInnerHTML={{ __html: props.text }} />;
};

Text.propTypes = {
    tutorial: PropTypes.string.isRequired
}
export default Text;
