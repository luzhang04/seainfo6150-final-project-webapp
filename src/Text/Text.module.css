.content {
  text-align: justify;
  background: #edf6f9;
  border: 2px dashed #edf6f9;
  padding: 10px;
  box-sizing: border-box;

}


