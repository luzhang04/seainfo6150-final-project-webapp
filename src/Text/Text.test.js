import React from "react";
import { render } from "@testing-library/react";
import Text from "./Text.jsx";

describe("Text tests", () => {
    it("renders correctly", () => {
      const { container } = render(<Text text="the text"/>);
      expect(container).toMatchSnapshot();
    });
  });