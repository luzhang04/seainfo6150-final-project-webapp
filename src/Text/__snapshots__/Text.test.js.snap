// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text tests renders correctly 1`] = `
<div>
  <div
    class="content"
  >
    the text
  </div>
</div>
`;
